File: 1.1/src/cz/mg/vulkan/PFNvkCmdWriteTimestamp.java
package cz.mg.vulkan;

public class PFNvkCmdWriteTimestamp extends VkFunctionPointer {
    public PFNvkCmdWriteTimestamp() {
    }

    protected PFNvkCmdWriteTimestamp(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdWriteTimestamp(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdWriteTimestamp(long value) {
        setValue(value);
    }

    public PFNvkCmdWriteTimestamp(VkInstance instance) {
        super(instance, new VkString("vkCmdWriteTimestamp"));
    }

    public void call(VkCommandBuffer commandBuffer, VkPipelineStageFlagBits pipelineStage, VkQueryPool queryPool, VkUInt32 query){
        call(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), pipelineStage != null ? pipelineStage.getVkAddress() : VkPointer.getNullAddressNative(), queryPool != null ? queryPool.getVkAddress() : VkPointer.getNullAddressNative(), query != null ? query.getVkAddress() : VkPointer.getNullAddressNative());
    }

    protected static native void call(long vkaddress, long commandBuffer, long pipelineStage, long queryPool, long query);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkCmdWriteTimestamp.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdWriteTimestamp_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jint pipelineStage, jlong queryPool, jint query){
    (void)env;
    (void)jc;
    PFN_vkCmdWriteTimestamp f = (PFN_vkCmdWriteTimestamp)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        (VkPipelineStageFlagBits)pipelineStage,
        *((VkQueryPool*)jniLongToPointer(queryPool)),
        (uint32_t)query
    );
}

